frontend/javascripts/oxalis/model/sagas/isosurface_saga.js

```diff
@@ -369,6 +374,172 @@ function* maybeLoadIsosurface(
+    // This is done to ensure that every voxel of the isosurface is reloaded.
+    yield* call(ensureSuitableIsosurface, position, cellId, shouldBeRemoved);
+    shouldBeRemoved = false;
+  }
+  yield* put(finishedLoadingIsosurfaceAction(layerName, cellId));
+}
+
+/*
+ *
+ * Precomputed Meshes
+ *
+ */
+
+function* loadPrecomputedMesh(action: LoadPrecomputedMeshAction) {
+  const { cellId, seedPosition, meshFileName, layerName } = action;
+  const layer = yield* select(state =>
+    layerName != null
+      ? getSegmentationLayerByName(state.dataset, layerName)
+      : getVisibleSegmentationLayer(state),
+  );
+  if (layer == null) return;
+
+  // If a REMOVE_ISOSURFACE action is dispatched and consumed
+  // here before loadPrecomputedMeshForSegmentId is finished, the latter saga
+  // should be canceled automatically to avoid populating mesh data even though
+  // the mesh was removed. This is accomplished by redux-saga's race effect.
+  yield* race({
+    loadPrecomputedMeshForSegmentId: _call(
+      loadPrecomputedMeshForSegmentId,
+      cellId,
+      seedPosition,
+      meshFileName,
+      layer,
+    ),
+    cancel: _take(
+      otherAction =>
+        otherAction.type === "REMOVE_ISOSURFACE" &&
+        otherAction.cellId === cellId &&
+        otherAction.layerName === layer.name,
+    ),
+  });
+}
+
+function* loadPrecomputedMeshForSegmentId(
+  id: number,
+  seedPosition: Vector3,
+  fileName: string,
+  segmentationLayer: APIDataLayer,
+): Saga<void> {
+  const layerName = segmentationLayer.name;
+  yield* put(addIsosurfaceAction(layerName, id, seedPosition, true));
+  yield* put(startedLoadingIsosurfaceAction(layerName, id));
+  const dataset = yield* select(state => state.dataset);
+
+  let availableChunks = null;
+  try {
+    availableChunks = yield* call(
+      getMeshfileChunksForSegment,
+      dataset.dataStore.url,
+      dataset,
+      getBaseSegmentationName(segmentationLayer),
+      fileName,
+      id,
+    );
+  } catch (exception) {
+    console.warn("Mesh chunk couldn't be loaded due to", exception);
+    Toast.warning(messages["tracing.mesh_listing_failed"]);
+
+    yield* put(finishedLoadingIsosurfaceAction(layerName, id));
+    yield* put(removeIsosurfaceAction(layerName, id));
+    return;
+  }
+
+  // Sort the chunks by distance to the seedPosition, so that the mesh loads from the inside out
+  const sortedAvailableChunks = _.sortBy(availableChunks, chunkPosition =>
+    V3.length(V3.sub(seedPosition, chunkPosition)),
+  );
+  const tasks = sortedAvailableChunks.map(
+    chunkPosition =>
+      function* loadChunk() {
+        const stlData = yield* call(
+          getMeshfileChunkData,
+          dataset.dataStore.url,
+          dataset,
+          getBaseSegmentationName(segmentationLayer),
+          fileName,
+          id,
+          chunkPosition,
+        );
+
+        const geometry = parseStlBuffer(stlData);
+        getSceneController().addIsosurfaceFromGeometry(geometry, id);
```

Comment:
It probably doesn't make a real difference, since the parsing is synchronous, but using `call` for the two lines, would be more consistent, I guess.